subiquity/server/controllers/ad.py

```diff
@@ -95,12 +97,19 @@ class ADController(SubiquityController):
 
     def __init__(self, app):
         super().__init__(app)
+        self.ad_joiner = AdJoiner(self.app)
+        self.join_result = AdJoinResult.UNKNOWN
         if self.app.opts.dry_run:
             self.ping_strgy = StubDcPingStrategy()
         else:
             self.ping_strgy = DcPingStrategy()
 
     def start(self):
+        # AD is by nature an optional feature.
+        # Yet, we need to make its model part of the POSTINSTALL_MODEL_NAMES
+        # set, which would turn this into a required controller.
+        # So, let's start this as configured by default.
+        run_bg_task(self.configured())
```

Comment:
I think you don't want this, and the client should POST to mark it configured instead.
The defect would be, the user intends to use AD, they do most of the install but let the installer screen sit on the AD page.  The `postinstall` method in the `install` controller runs, no AD join is attempted, and then the user inputs join information that won't be processed.
Instead, the client should POST empty connection info if AD is undesired, and that means that before that POST is done, the `postinstall` method will be blocked from running.